snap/snapenv/snapenv.go

```diff
@@ -52,37 +44,35 @@ const (
 //
 // With the extra parameter additional environment variables can be
 // supplied which will be set in the execution environment.
-func ExecEnv(info *snap.Info, extra map[string]string) []string {
-	// merge environment and the snap environment, note that the
-	// snap environment overrides pre-existing env entries
-	preserve := discardUnsafeFlag
-	if info.NeedsClassic() {
-		preserve = preserveUnsafeFlag
+func ExecEnv(info *snap.Info) (osutil.Environment, error) {
+	// Start with OS environment.
+	env, err := osutil.OSEnvironment()
```

Comment:
this probably should change and not find the os enviroment on its own, tests would also be simpler

that was not possible before because the preserving logic was also here